docs/user/user/adding-glean-to-your-project/javascript.md

```diff
@@ -0,0 +1,133 @@
+# Adding Glean to your Javascript project
+
+Javascript projects use the Javascript APIs provided by Glean.js.
+
+Currently, these bindings support collecting data from [Browser Extensions](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions) (cross-browser) and [Qt/QML applications](https://doc.qt.io/qt-5/qtqml-index.html).
+
+## Requirements
+
+* Node.js >= 12.20.0.
+* npm >= 7.0.0.
+
+## Setting up the dependency
+
+The Glean.js package is distributed as an npm package [`@mozilla/glean`](https://www.npmjs.com/package/@mozilla/glean).
+
+Install Glean.js in your Javascript project, by running:
+
+```bash
+npm install @mozilla/glean
+```
+
+Then import Glean into your project:
+
+```js
+// Glean.js for webextensions
+//
+// esm
+import Glean from "@mozilla/glean/webext";
+// cjs
+const { default: Glean } = require("@mozilla/glean/webext");
+```
+
+### Common import errors
+
+#### "Cannot find module '@mozilla/glean'"
+
+Glean.js does not have a [`main`](https://nodejs.org/api/packages.html#packages_main_entry_point_export) package entry point.
+Instead it relies on a series of entry points depending on the platform you are targeting.
+ 
+In order to import Glean use:
+
+```js
+import Glean from '@mozilla/glean/{your-platform}'
+```
+ 
+#### "Module not found: Error: Can't resolve '@mozilla/glean/webext' in '...'"
+ 
+Glean.js relies on Node.js' [subpath exports](https://nodejs.org/api/packages.html#packages_subpath_exports)
+feature to define multiple package entry points.
+ 
+Please make sure that you are using a supported Node.js runtime and also make sure the tools you are using support this Node.js feature.
```

Comment:
I wonder if we should mention specifically that Webpack v4 does not support it. We will never have a fully comprehensive list of tools that don't support subpath exports, but Webpack is large enough and it might be nice to mention it.

Also, this information living here, makes this page redundant: https://github.com/mozilla/glean.js/blob/main/docs/troubleshooting.md (which is good, it's best this information live here IMO). Would you mind filing a follow up to remove that page though?

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1713571